### ISSUE TITLE
[Enhancement] Let Normalize clip its output to an explicit range

### ISSUE THREAD

**Issue by ternaus** [albumentations-team/AlbumentationsX#364]
An AlbumentationsX adopter has a 16-channel hyperspectral pipeline that normalizes with dataset-level per-channel minima and maxima and then clips the result. The fixed affine part already fits [`Normalize`](https://albumentations.ai/explore/transform/Normalize/), but the clipping does not.

The project currently stores fixed bounds in its config:

```yaml
global_max: [0.6203158, 0.6172642, 0.46794897, 0.4325111, 0.4996644, 0.61997396,
             0.7382196, 0.86097705, 0.88304037, 0.9397393, 1.1892519, 1.5035477,
             1.4947973, 1.4737314, 1.6318618, 1.7226081]
global_min: [0.00028473, 0.0043945, 0.00149752, 0.00167517, 0.00190101, 0.0028114,
             0.00394378, 0.00488099, 0.00257091, 0.00215704, 0.00797662, 0.01205248,
             0.01310135, 0.01476806, 0.01932094, 0.02020744]
range_mode: "0_1"
```

Source: [`hsi_dermoscopy_croppedv2.yaml`](https://github.com/teaching-FEEC/dgm-2025.2/blob/46e5b6a001c1615521bdfb287e58c8b3f525e510/projects/hyperskin/configs/data/hsi_dermoscopy_croppedv2.yaml#L15-L21)

It implements the operation in a custom `ImageOnlyTransform`:

```python
denom = maxs - mins
denom = torch.where(denom == 0, torch.tensor(1e-8, device=device), denom)

norm_img = (img - mins[:, None, None]) / denom[:, None, None]

if self.range_mode == "-1_1":
    norm_img = norm_img * 2.0 - 1.0

if self.clip:
    min_val = -1.0 if self.range_mode == "-1_1" else 0.0
    norm_img = torch.clamp(norm_img, min_val, 1.0)
```

Source: [`NormalizeByMinMax.apply`](https://github.com/teaching-FEEC/dgm-2025.2/blob/46e5b6a001c1615521bdfb287e58c8b3f525e510/projects/hyperskin/src/transforms/norm_min_max.py#L63-L81)

The data module constructs and appends that transform to all three pipelines:

```python
norm_transform = NormalizeByMinMax(
    mins=self.global_min,
    maxs=self.global_max,
    range_mode=self.range_mode,
    clip=True,
)

for t in ["train", "val", "test"]:
    comp = getattr(self, f"transforms_{t}")
    if comp is not None:
        comp.transforms.append(norm_transform)
```

Source: [`BaseDataModule._build_transforms`](https://github.com/teaching-FEEC/dgm-2025.2/blob/46e5b6a001c1615521bdfb287e58c8b3f525e510/projects/hyperskin/src/data_modules/base.py#L107-L131)

## What is already expressible

For output in `[0, 1]`, fixed per-channel min-max scaling is standard normalization with:

```python
mean = mins
std = maxs - mins
max_pixel_value = 1.0
```

For output in `[-1, 1]`, it is:

```python
mean = (mins + maxs) / 2
std = (maxs - mins) / 2
max_pixel_value = 1.0
```

This is different from `normalization="min_max_per_channel"`, which derives new extrema from every input image instead of using fixed dataset statistics.

I checked both formulas against the custom transform with the published `albumentationsx==2.3.6` package on a 16-channel float32 image. Values inside the configured bounds matched within `2.4e-7`. Values outside the bounds expose the missing contract: the custom output remained in `[0, 1]` or `[-1, 1]`, while [`Normalize`](https://albumentations.ai/explore/transform/Normalize/) correctly continued the affine mapping beyond those intervals.

## Proposed API

Add an optional explicit output range to [`Normalize`](https://albumentations.ai/explore/transform/Normalize/):

```python
A.Compose([
    *spatial_transforms,
    A.Normalize(mean=mean, std=std, max_pixel_value=1.0, clip_range=(0.0, 1.0)),
    A.ToTensorV2(),
])
```

Suggested contract:

```python
clip_range: tuple[float, float] | None = None
```

- `None` preserves current behavior.
- A tuple clips after the selected normalization mode.
- The argument is validated and serialized like the other constructor arguments.
- The same behavior applies to `image`, `images`, and `volume`.

The adopter's custom class also tries to infer whether an input is already normalized and bypasses the fixed statistics. I would not add that inference here: raw scientific images can naturally lie in `[0, 1]`, so input state should remain explicit.

## Acceptance checks

- Existing [`Normalize`](https://albumentations.ai/explore/transform/Normalize/) calls remain byte-for-byte unchanged when `clip_range=None`.
- Standard normalization supports scalar and per-channel statistics, including more than four channels, followed by clipping.
- Tests cover float32 and uint8 input, `[0, 1]` and `[-1, 1]` output ranges, plural images, 3D data, serialization, and invalid/reversed bounds.
- Documentation distinguishes fixed dataset bounds from per-image `min_max` and `min_max_per_channel` modes.

I searched the open and closed AlbumentationsX and legacy Albumentations trackers for fixed/dataset min-max normalization and output clipping before opening this; I did not find an existing request with this contract.
